control-plane/plugin/src/resources/node.rs

```diff
@@ -118,23 +118,58 @@ fn drain_labels_from_state(ds: &CordonDrainState) -> Vec<String> {
 impl Cordoning for Node {
     type ID = NodeId;
     async fn cordon(id: &Self::ID, label: &str, output: &OutputFormat) {
-        match RestClient::client()
-            .nodes_api()
-            .put_node_cordon(id, label)
-            .await
-        {
-            Ok(node) => match output {
-                OutputFormat::Yaml | OutputFormat::Json => {
-                    // Print json or yaml based on output format.
-                    utils::print_table(output, node.into_body());
+        // is node already cordoned with the label?
+        let already_has_cordon_label: bool =
+            match RestClient::client().nodes_api().get_node(id).await {
+                Ok(node) => {
+                    let node_body = &node.into_body();
+
+                    match &node_body.spec.as_ref().unwrap().cordondrainstate {
+                        Some(ds) => cordon_labels_from_state(ds).contains(&label.to_string()),
+                        None => false,
+                    }
                 }
-                OutputFormat::None => {
-                    // In case the output format is not specified, show a success message.
-                    println!("Node {} cordoned successfully", id)
+                Err(e) => {
+                    println!("Failed to get node {}. Error {}", id, e);
+                    return;
+                }
+            };
+        if !already_has_cordon_label {
```

Comment:
the code inside these two matches seem to be exactly the same? If so then I'd suggest reusing it to avoid duplication, eg:
```rust
let result = match already_has_cordon_label {
  true => RestClient::client().nodes_api().put_node_cordon(id, label).await,
  false => RestClient::client().nodes_api().get_node(id).await,
};
match result {
  Ok(node) => match output {
      OutputFormat::Yaml | OutputFormat::Json => {
          // Print json or yaml based on output format.
          utils::print_table(output, node.into_body());
      }
      OutputFormat::None => {
          // In case the output format is not specified, show a success message.
          println!("Node {} cordoned successfully", id)
      }
  },
  Err(e) => {
      println!("Failed to cordon node {}. Error {}", id, e)
  }
}
```

Reply:
fixed